karate-junit5/src/main/java/com/intuit/karate/junit5/Karate.java

```diff
@@ -60,6 +61,8 @@ public static Karate run(String... paths) {
         List<DynamicNode> list = new ArrayList();
         List<CompletableFuture> futures = new ArrayList();
         for (Feature feature : suite.features) {
+            if (!FeatureRuntime.of(suite, feature).scenarios.hasNext()) // if there is no scenarios selected for execution, skip the feature
```

Comment:
nope. this will be done again in `FeatureNode` as you can see. please figure out a way to init the runtime only once and pass it around / redesign `FeatureNode` if needed

Reply:
I see. will be back with next try.

Reply:
Thanks for the hint! `FeatureNode` does already has a method hasNext(). Now I do not init the runtime.

Reply:
perfect ! thanks